src/mtcp/mtcp_restart.c

```diff
@@ -1232,12 +1231,12 @@ remapMtcpRestartToReservedArea(RestoreInfo *rinfo)
   VA guard_page =
     mem_regions[num_regions - 1].endAddr + restore_region_offset;
 
-  MTCP_ASSERT(mtcp_sys_mmap(guard_page,
-                            MTCP_PAGE_SIZE,
-                            PROT_NONE,
-                            MAP_ANONYMOUS | MAP_PRIVATE | MAP_FIXED,
-                            -1,
-                            0) == guard_page);
+  MTCP_ASSERT(mmap_fixed_noreplace(guard_page,
+                                   MTCP_PAGE_SIZE,
+                                   PROT_NONE,
```

Comment:
REPEATED:
The `MAP_FIXED` is redundant, since `map_fixed_noreplace()` will simply remove the `MAP_FIXED`.
We can keep it here, or remove `MAP_FIXED` with the idea that the name `map_fied_noreplace()` already says it all.
I'm open to either option (keeping it as is, or removing it).  Just a thought.